WooCommerce/Classes/Extensions/String+HTML.swift

```diff
@@ -1,10 +1,18 @@
 import Foundation
-
+import class Aztec.HTMLParser
 
 /// String: HTML Stripping
 ///
 extension String {
 
+    /// Returns the HTML Stripped version of the receiver.
+    ///
+    /// NOTE: I can be very slow ⏳ — using it in a background thread is strongly recommended.
+    ///
+    var strippedHTML: String {
+        HTMLParser().parse(self).rawText()
+    }
```

Comment:
I ran into this situation too when scraping with Python, especially in the case where wp-login.php was redirected to the home page. Some sites have a very big home page and the HTML can be really long with tens of thousands characters.

My workaround was to check if the string is longer than some 5000 characters. if it is, it's probably a homepage redirect situation since error pages usually aren't that long, then I skip parsing the HTML. It's not perfect, but helps cut down the processing time.

Reply:
Currently, I'm only stripping HTML when I find a matching `div` for the login error messaging, which is a small div and should be fast.

Reply:
I'll add also another point, we don't scrap the HTML when there is redirection, only when we get a 200 response from the login page.